src/main/java/com/amazon/opendistroforelasticsearch/sql/query/ESActionFactory.java

```diff
@@ -94,6 +98,7 @@ public static QueryAction create(Client client, String sql) throws SqlParseExcep
                             .parseMultiSelect((SQLUnionQuery) sqlExpr.getSubQuery().getQuery());
                     return new MultiQueryAction(client, multiSelect);
                 } else if (isJoin(sqlExpr, sql)) {
+                    new JoinRewriteRule(LocalClusterState.state()).rewrite(sqlExpr);
```

Comment:
Have you explor the possiblity to combine this with SubqueryAliasRewriter?

Reply:
No, I haven't explored that possibility.  I wanted separation of concerns and did not wanted to convolute the code. Will check  if we can combine this cleanly with `SubqueryAliasRewriter`?